src/gpu_util.cpp

```diff
@@ -269,19 +269,68 @@ void gpu_get_all_domain_id(std::vector<uint16_t>* pgpus_domain_id,
  * @param device_id GPU Device ID
  * @return true if GPU is die in MCM GPU, false if GPU is single die GPU.
  **/
-
-
 bool gpu_check_if_mcm_die (int idx) {
   rsmi_status_t ret;
   uint64_t val =0 , time_stamp;
   float cntr_resolution;
+  uint32_t smi_index = 0;
+
+  if (gpu_hip_to_smi_index(idx, &smi_index)) {
+    return false;
+  }
+
   // in case of secondary die, energy accumulator will return zero. 
-  ret = rsmi_dev_energy_count_get(idx, &val, &cntr_resolution, &time_stamp);
+  ret = rsmi_dev_energy_count_get(smi_index, &val, &cntr_resolution, &time_stamp);
   if (!((RSMI_STATUS_SUCCESS == ret) && val == 0))
-	  return false;
+    return false;
   return true;
 }
 
+/**
+ * @brief Get GPU smi index from hip index.
+ * @param hip_index GPU hip index
+ * @param smi_index GPU smi index
+ * @return 0 if successful, -1 otherwise
+ **/
+int gpu_hip_to_smi_index(int hip_index, uint32_t* smi_index) {
+
+  int hip_num_gpu_devices = 0;
+  uint32_t smi_num_devices = 0;
+  uint64_t val_ui64 = 0;
+  std::map<uint64_t, int> smi_map;
```

Comment:
this can atleast be made a singleton/static to avoid repeat runs .. Approving for release block, note as todo